RealmSwift/Realm.swift

```
@@ -223,55 +227,68 @@ public final class Realm {
 
-        RLMAddObjectToRealm(object, rlmRealm, RLMCreationOptions.UpdateOrCreate)
-    }
+    Creates or updates an instance of this object and adds it to the `Realm` populating
+    the object represented by value.
+    
+    When 'update' is 'true', the object must have a primary key. If no objects exist in
+    the Realm instance with the same primary key value, the object is inserted. Otherwise, 
+    the existing object is updated with any changed values.
 
-    /**
-    Adds or updates objects in the given array to be persisted it in this Realm.
+    :param: type    The object type to create.
+    :param: value   The value used to populate the object. This can be any key/value coding compliant
+                    object, or a JSON dictionary such as those returned from the methods in `NSJSONSerialization`,
+                    or an `Array` with one object for each persisted property. An exception will be
+                    thrown if any required properties are not present and no default is set.
 
-    :see: addOrUpdate(object:)
+                    When passing in an `Array`, all properties must be present,
+                    valid and in the same order as the properties defined in the model.
+    :param: update  If true will try to update existing objects with the same primary key.
 
-    :param: objects A sequence of `Object`s to be added to this Realm.
+    :returns: The created object.
     */
-    public func addOrUpdate<S: SequenceType where S.Generator.Element: Object>(objects: S) {
-        for obj in objects {
-            addOrUpdate(obj)
+    public func create<T: Object>(type: T.Type, value: AnyObject = [:], update: Bool = false) -> T {
+        var options : RLMCreationOptions = .allZeros
+        if update == true {
+            options = RLMCreationOptions.UpdateOrCreate
+            if schema[T.className()]?.primaryKeyProperty == nil {
+                fatalError("'\(T.className())' does not have a primary key and can not be updated")
```

All `fatalError`s should throw an exception instead for testing purposes.